### PR TITLE
Only merge system defaults when querying workflow execution config for project attributes

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
The project-level-only matchable attribute endpoint currently merges in defaults from the server config (as this was the quickest way to get things going in light of the need to expose this to the UI and the complexity of the broader settings project).  There was a bug in the [initial implementation](https://github.com/flyteorg/flyteadmin/pull/480) though - these system level defaults should only be merged when the user is querying for workflow execution configs.  This just adds a check for that.

See the original PR for more info as well.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Also update some the default config.yaml to work with the new single-binary based demo sandbox.

## Tracking Issue
Just noticed this in testing.

### CODE REVIEW

**Review by katrogan — APPROVED** [review]
Is it worth adding a unit test?

**Review by katrogan — DISMISSED** [review]

**Review by katrogan — APPROVED** [review]
